containers/src/Kokkos_DynamicView.hpp

```diff
@@ -267,10 +267,18 @@ class DynamicView : public Kokkos::ViewTraits<DataType, P...> {
 
  public:
   //----------------------------------------------------------------------
+  using uniform_type =
+      DynamicView<typename traits::data_type, typename traits::device_type>;
+
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  /** \brief  Compatible view of data type */
+  using type KOKKOS_DEPRECATED_WITH_COMMENT("Use uniform_type instead.") =
+      uniform_type;
+#endif
 
   /** \brief  Compatible view of array of scalar types */
-  using array_type =
-      DynamicView<typename traits::data_type, typename traits::device_type>;
+  using array_type KOKKOS_DEPRECATED_WITH_COMMENT("Use uniform_type instead.") =
+      uniform_type;
```

Comment:
If it's depreacted, it shouldn't exist when `Kokkos_ENABLE_DEPRECATED_COE_4=OFF` is set.